### PR TITLE
Booking widget shows the turnaround

### PR DESCRIPTION
**Pull request by bryangaledxw**
# Changes in this PR

This PR updates the app to properly show closed bookings, and to show the turnaround time for bookings. With the exception of some small text changes, this feature is hidden behind the `disbaledTurnarounds` feature flag

### Before
localhost_3000_properties_d86befab-2bef-4e89-b540-16986cbf7dbc_bedspaces_16e112dc-ccf5-4c48-8c51-8f1ff5f60a91_bookings_33c778a9-d941-4c64-88ad-d343314b357c (1)

### After
localhost_3000_properties_d86befab-2bef-4e89-b540-16986cbf7dbc_bedspaces_16e112dc-ccf5-4c48-8c51-8f1ff5f60a91_bookings_33c778a9-d941-4c64-88ad-d343314b357c

# Release checklist

[Release process
documentation](https://dsdmoj.atlassian.net/wiki/spaces/AP/pages/edit-v2/4247847062?draftShareId=a1c360ab-bd31-4db1-aae3-7cc002761de9).

## Pre-merge checklist

- [ ] Are any changes required to dependent services for this change to work?
  (eg. CAS API)
- [ ] Have they been released to production already?

## Post-merge checklist

- [ ] [Manually
  approve](https://dsdmoj.atlassian.net/wiki/spaces/AP/pages/4247847062/Release+process#Manual-releases)
  release to test
- [ ] [Manually
  approve](https://dsdmoj.atlassian.net/wiki/spaces/AP/pages/4247847062/Release+process#Manual-releases)
  release to preprod
- [ ] [Manually
  approve](https://dsdmoj.atlassian.net/wiki/spaces/AP/pages/4247847062/Release+process#Manual-releases)
  release to prod

### CODE REVIEW

**Comment by bryangaledxw** [comment]
[Booking widget shows the turnaround](https://trello.com/c/bFSGsuRp/1021-booking-widget-shows-the-turnaround)

File: server/utils/bookingUtils.ts [hunk 1]
```diff
@@ -104,10 +104,15 @@ export const allStatuses: Array<{ name: string; id: Booking['status']; tagClass:
     tagClass: 'govuk-tag--green',
   },
   {
-    name: 'Departed',
+    name: config.flags.turnaroundsDisabled ? 'Departed' : 'Departed - Turnaround In Progress',
     id: 'departed',
     tagClass: 'govuk-tag--red',
   },
+  {
+    name: 'Departed',
```
Comment (patrickjfl) [inline server/utils/bookingUtils.ts:114]: Do we not want a condition on this so it's 'Closed' if turnarounds are enabled?

File: cypress_shared/pages/temporary-accommodation/manage/bedspaceShow.ts [hunk 2]
```diff
@@ -66,6 +66,10 @@ export default class BedspaceShowPage extends Page {
         } else if (status === 'arrived') {
           cy.get('td').eq(3).contains('Active')
         } else if (status === 'departed') {
+          cy.get('td')
+            .eq(3)
+            .contains(Cypress.env('testLevel') === 'e2e' ? 'Departed' : 'Departed - Turnaround In Progress')
```
Comment (patrickjfl) [inline cypress_shared/pages/temporary-accommodation/manage/bedspaceShow.ts]: This seems neat but I'm also a bit confused by it. It's the obvious question but are we sure that in E2E tests a 'departed' status will correspond to a 'Departed' UI status?

File: server/components/bookingInfo.ts [hunk 3]
```diff
@@ -42,6 +43,21 @@ export default (booking: Booking): SummaryList['rows'] => {
...
+    const days = booking.turnaround.workingDays
+
+    rows.push(
+      {
+        key: textValue('Turnaround time'),
+        value: textValue(`${days} working ${days === 1 ? 'day' : 'days'}`),
+      },
+      {
+        key: textValue('Turnaround end date'),
+        value: textValue(DateFormats.isoDateToUIDate(booking.effectiveEndDate || booking.departureDate)),
```
Comment (patrickjfl) [inline server/components/bookingInfo.ts:56]: When do we use the departureDate?

File: server/services/bookingService.ts [hunk 4]
```diff
@@ -54,7 +54,14 @@ export default class BookingService {
         rows: [
           this.textValue(b.person.crn),
           this.textValue(DateFormats.isoDateToUIDate(b.arrivalDate, { format: 'short' })),
-          this.textValue(DateFormats.isoDateToUIDate(b.departureDate, { format: 'short' })),
+          this.textValue(
+            DateFormats.isoDateToUIDate(
+              config.flags.turnaroundsDisabled ? b.departureDate : b.effectiveEndDate || b.departureDate,
```
Comment (patrickjfl) [inline server/services/bookingService.ts:59]: same question here about departureDate

File: server/utils/bookingUtils.ts [hunk 5]
```diff
@@ -104,10 +104,15 @@ export const allStatuses: Array<{ name: string; id: Booking['status']; tagClass:
     tagClass: 'govuk-tag--green',
   },
   {
-    name: 'Departed',
+    name: config.flags.turnaroundsDisabled ? 'Departed' : 'Departed - Turnaround In Progress',
```
Comment (patrickjfl) [inline server/utils/bookingUtils.ts]: I ran this by Selina (cos capitalisation is a bugbear of mine) but she says 'Departed - turnaround in progress' should now just say 'Turnaround'

**Review by patrickjfl — APPROVED** [review]
There's a lot here and I think the task itself is complex so I'm slightly confused here and there, but on the whole I think it looks good. Nice work.

File: server/utils/bookingUtils.ts (hunk 1 above)
Comment (bryangaledxw) [reply server/utils/bookingUtils.ts:114]: We're not presenting a 'Closed' status to the user, the status are/were showing 'Departed - Turnaround In Progess' (what we were showing as 'Departed' corresponding to `departed`), and 'Departed' (corresponding to `closed`)

File: server/components/bookingInfo.ts (hunk 3 above)
Comment (bryangaledxw) [reply server/components/bookingInfo.ts:56]: When the effectiveEndDate is null - when either the booking doesn't have a turnaround, or it has a zero-day turnaround

File: server/utils/bookingUtils.ts (hunk 5 above)
Comment (bryangaledxw) [reply server/utils/bookingUtils.ts]: Will chase - this has knock-on UI implications for the booking search

File: cypress_shared/pages/temporary-accommodation/manage/bedspaceShow.ts (hunk 2 above)
Comment (bryangaledxw) [reply cypress_shared/pages/temporary-accommodation/manage/bedspaceShow.ts]: My thinking here was that 'Departed' will match both 'Departed' and 'Departed - Turnaround In Progress'. Will need to rethink with it changing to 'Turnaround'

File: server/utils/bookingUtils.ts (hunk 1 above)
Comment (patrickjfl) [reply server/utils/bookingUtils.ts:114]: Ahh I see

File: cypress_shared/pages/temporary-accommodation/manage/bedspaceShow.ts (hunk 2 above)
Comment (patrickjfl) [reply cypress_shared/pages/temporary-accommodation/manage/bedspaceShow.ts]: ok 👍

File: server/components/bookingInfo.ts (hunk 3 above)
Comment (patrickjfl) [reply server/components/bookingInfo.ts:56]: thanks 👍

**Pull request merged**